Auxiliary/DirectXTexEXR.cpp

```diff
@@ -386,8 +396,17 @@ HRESULT DirectX::LoadFromEXRFile(const wchar_t* szFile, TexMetadata* metadata, S
             metadata->dimension = TEX_DIMENSION_TEXTURE2D;
         }
 
-        hr = image.Initialize2D(DXGI_FORMAT_R16G16B16A16_FLOAT,
-            static_cast<size_t>(width), static_cast<size_t>(height), 1u, 1u);
+        if (!isEnvmap)
+        {
+            hr = image.Initialize2D(DXGI_FORMAT_R16G16B16A16_FLOAT,
+                static_cast<size_t>(width), static_cast<size_t>(height), 1u, 1u);
+        }
+        else
+        {
+            hr = image.InitializeCube(DXGI_FORMAT_R16G16B16A16_FLOAT,
```

Comment:
Within the context of the command-line tools, it should probably be an array and not a cube. I can add behavior flags to allow it to do either, or I can just make it always load as a texture array. I can then add EXR support to ``texassemble`` to support EXR to cubemap DDS conversions.

Reply:
IOW: I suggest this code be:

```
hr = image.Initialize2D(DXGI_FORMAT_R16G16B16A16_FLOAT,
            static_cast<size_t>(width), static_cast<size_t>(height), (isEnvmap) ? 6u : 1u, 1u);
```

Reply:
Fair enough, updated. Keep in mind that the code was tested before the final changes as loading and initializing cubemap texture for a DXR Miss Shader